api/src/main/java/org/apache/cloudstack/api/command/admin/config/ResetCfgCmd.java

```diff
@@ -0,0 +1,164 @@
+    private Long clusterId;
+
+    @Parameter(name = ApiConstants.STORAGE_ID,
+               type = CommandType.UUID,
+               entityType = StoragePoolResponse.class,
+               description = "the ID of the Storage pool to reset the parameter value for corresponding storage pool")
+    private Long storagePoolId;
+
+    @Parameter(name = ApiConstants.DOMAIN_ID,
+            type = CommandType.UUID,
+            entityType = DomainResponse.class,
+            description = "the ID of the Domain to reset the parameter value for corresponding domain")
+    private Long domainId;
+
+    @Parameter(name = ApiConstants.ACCOUNT_ID,
+               type = CommandType.UUID,
+               entityType = AccountResponse.class,
+               description = "the ID of the Account to reset the parameter value for corresponding account")
+    private Long accountId;
+
+    @Parameter(name = ApiConstants.IMAGE_STORE_UUID,
+            type = CommandType.UUID,
+            entityType = ImageStoreResponse.class,
+            description = "the ID of the Image Store to reset the parameter value for corresponding image store")
+    private Long imageStoreId;
+
+    /////////////////////////////////////////////////////
+    /////////////////// Accessors ///////////////////////
+    /////////////////////////////////////////////////////
+
+    public String getCfgName() {
+        return cfgName;
+    }
+
+    public Long getZoneId() {
+        return zoneId;
+    }
+
+    public Long getClusterId() {
+        return clusterId;
+    }
+
+    public Long getStoragepoolId() {
+        return storagePoolId;
+    }
+
+    public Long getDomainId() {
+        return domainId;
+    }
+
+    public Long getAccountId() {
+        return accountId;
+    }
+
+    public Long getImageStoreId() {
+        return imageStoreId;
+    }
+
+    /////////////////////////////////////////////////////
+    /////////////// API Implementation///////////////////
+    /////////////////////////////////////////////////////
+
+    @Override
+    public String getCommandName() {
+        return s_name;
+    }
+
+    @Override
+    public long getEntityOwnerId() {
+        return Account.ACCOUNT_ID_SYSTEM;
+    }
+
+    @Override
+    public void execute() {
+        Pair<Configuration, String> cfg = _configService.resetConfiguration(this);
+        if (cfg != null) {
+            ConfigurationResponse response = _responseGenerator.createConfigurationResponse(cfg.first());
+            response.setResponseName(getCommandName());
+            if (getZoneId() != null) {
+                response.setScope("zone");
+            }
+            if (getClusterId() != null) {
+                response.setScope("cluster");
+            }
+            if (getStoragepoolId() != null) {
+                response.setScope("storagepool");
+            }
+            if (getDomainId() != null) {
+                response.setScope("domain");
```

Comment:
Use scope defined in ConfigKey, ConfigKey.Scope.Domain.toString() here. Same for the other scopes values.